nexus/db-queries/src/db/pool_connection.rs

```diff
@@ -4,46 +4,91 @@
 
 //! Customization that happens on each connection as they're acquired.
 
+use anyhow::anyhow;
+use async_bb8_diesel::AsyncR2D2Connection;
 use async_bb8_diesel::AsyncSimpleConnection;
-use async_bb8_diesel::Connection;
-use async_bb8_diesel::ConnectionError;
 use async_trait::async_trait;
-use bb8::CustomizeConnection;
+use diesel::Connection;
 use diesel::PgConnection;
 use diesel_dtrace::DTraceConnection;
+use qorb::backend::{self, Backend, Error};
 
 pub type DbConnection = DTraceConnection<PgConnection>;
 
 pub const DISALLOW_FULL_TABLE_SCAN_SQL: &str =
     "set disallow_full_table_scans = on; set large_full_scan_rows = 0;";
 
-/// A customizer for all new connections made to CockroachDB, from Diesel.
-#[derive(Debug)]
-pub(crate) struct ConnectionCustomizer {}
+/// A [backend::Connector] which provides access to [PgConnection].
+pub struct DieselPgConnector {
+    prefix: String,
+    suffix: String,
+}
 
-impl ConnectionCustomizer {
-    pub(crate) fn new() -> Self {
-        Self {}
+impl DieselPgConnector {
+    /// Creates a new "connector" to a database, which
+    /// swaps out the IP address at runtime depending on the selected backend.
+    ///
+    /// Format of the url is:
+    ///
+    /// - postgresql://{user}@{address}/{db}
+    ///
+    /// Or, if arguments are supplied:
+    ///
+    /// - postgresql://{user}@{address}/{db}?{args}
+    pub fn new(user: &str, db: &str, args: Option<&str>) -> Self {
```

Comment:
nitpicky: a function taking multiple positional arguments of the same type with different semantics is always kind of a red flag to me...it seems easy to accidentally swap the ordering of the username and db. if this is going to be a public API, i might consider a struct which passes these arguments as named fields, or newtypes or something. on the other hand, if this is internal to the `db-queries` crate... :woman_shrugging:

Reply:
Updated to use a struct explicitly (even though it is, as you suggested, crate-internal)